### ISSUE TITLE
Optionally include URI in BigML locators

### ISSUE THREAD

**Issue by fwallacevt** [dbcrossbar/dbcrossbar#151]
Old:
`source:XXXXXXX`

New:
`source:bigml.io:XXXXXXX`
`source:my.bigml.uri.io:XXXXXXX`

If domain is missing, it is assumed to be `bigml.io` for backwards compatibility.

**Comment by skelliest** [comment]
Note: Although you can pass in the BigML domain+creds in the environment, that isn't the point of the request. It is important to be able to infer the target domain from just the URI, otherwise it isn't universal.

This also matches the way dbcrossbar handles bigquery URIs: `bigquery:project:object`, where `project` is analogous to the BigML domain here.

**Comment by emk** [comment]
Yeah, we need to deal with this. Please make sure that it gets scheduled through the usual process.

I'm likely to switch to a proper URI format with a host field, as in `bigml://my.bigml.uri.io/source/XXXXX`, but I'll keep compatibility with the old `bigml:source/XXXXX` format.

In general, BigQuery locators are weird, because we inherit their syntax from BigQuery, and because we want to be consistent with other BigQuery tools. But for non-BigQuery backends that have no established convention, I would prefer to use a standard URI-like syntax here.

**Comment by seamusabshere** [comment]
if you get a FQDN here, then perhaps you can solve https://github.com/dbcrossbar/dbcrossbar/issues/152 by checking the IP against a list of IPs for GCP vs AWS

**Comment by emk** [comment]
For now, this can be faked using `BIGML_DOMAIN` in the environment.
